gems/aws-sdk-core/lib/aws-sdk-core/rest/request/body.rb

```diff
@@ -17,11 +17,27 @@ def initialize(serializer_class, rules)
         # @param [Seahorse::Client::Http::Request] http_req
         # @param [Hash] params
         def apply(http_req, params)
-          http_req.body = build_body(params)
+          body = build_body(params)
+          # for rest-json, ensure we send at least an empty object
+          if body.nil? && @serializer_class == Json::Builder && modeled_body?
```

Comment:
Is there a reason this can't go in the `Json::Builder#serialize` (aka `to_json`) method.

`Json::Builder#serialize` is called only on the body - and the same runtime inspection of the model for body parameters is already done before calling it - so if its called with a non-empty set of params - we know a body is expected.

Does the empty body also apply to api-gateway?

Reply:
Talked offline - I don't think so unfortunately. The runtime inspection is an intersection of user params and body params, and is no guarantee that the operation can take a body.

Yes it applies to API Gateway. I've added the handler for it.